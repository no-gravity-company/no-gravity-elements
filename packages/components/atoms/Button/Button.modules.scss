@import '../../../stylesheets/base/variables';

$button-border-radius: var(--nge-button-border-radius, baseUnit(0.75));

@mixin button-styles {
  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: space-between;
  height: baseUnit(5);
  column-gap: baseUnit(1);
  padding: var(--nge-button-padding, baseUnit(1.25) baseUnit(2));
  border-radius: $button-border-radius;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  transition: var(--nge-button-transition, #{background-color 0.2s ease, color 0.2s ease});
  border: var(--nge-button-border, none);
}

@mixin button-hover($bg: red) {
  &:hover {
    background-color: $bg;
  }
}
@mixin button-active($bg: red, $textColor: #{$background}) {
  &:active {
    background-color: $bg;
    --text-color: #{$textColor};
  }
}
  .nge-button {
    width: var(--nge-button-width, auto);
    font-size: var(--nge-button-font-size, 16px);
    align-items: center;
    justify-content: center;
    font-weight: $font-weight-button;
    color: var(--text-color);
    background: unset;
    @include button-styles();

    &.disabled {
      cursor: default;
      pointer-events: none;
      opacity: .5;
    }


  &.primary { 
    @include button-hover($primary-hover);
    @include button-active($primary-active);
    background-color: $primary;
    color: var(--text-color, white);
  }

  &.disruptive {
    @include button-hover($disruptive-hover);
    @include button-active($disruptive-active);
    background-color: $disruptive;
    color: var(--text-color, white);
  }

  &.destructive {
    @include button-hover($destructive-hover);
    @include button-active($destructive-active);
    background-color: $destructive;
    color: var(--text-color, white);
  }

  &.outline {
    @include button-hover($primary-light);
    @include button-active($primary-light-active, $primary-hover);
    color: var(--text-color, #{$primary});
    --nge-button-border: 1px solid #{$primary};
    background-color: $background;
  }

  &.ghost {
    @include button-hover($primary-light);
    @include button-active($primary-light-active, $primary-hover);
    color: var(--text-color, #{$primary});
    background-color: $background;
  }
}


