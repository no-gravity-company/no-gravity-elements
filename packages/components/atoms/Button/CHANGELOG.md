# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [2.1.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@2.0.0...@no-gravity-elements/button@2.1.0) (2024-11-12)

### Features

- add lightDom web components ([7dff35f](https://github.com/no-gravity-company/no-gravity-elements/commit/7dff35f80579ff03893829a5a4f0cab415d6daf8))
- button with slot for html button ([86963a3](https://github.com/no-gravity-company/no-gravity-elements/commit/86963a3c0b1623b50e02fd05007c8c57cca7e975))
- new button styles ([c850785](https://github.com/no-gravity-company/no-gravity-elements/commit/c850785418de3a4f9ca393c75260ed407061314a))

# [2.0.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@1.2.0...@no-gravity-elements/button@2.0.0) (2024-04-03)

### Bug Fixes

- Strengthen event api typing ([2863217](https://github.com/no-gravity-company/no-gravity-elements/commit/286321767818f31a4002325d4bc5573e8996b080))

- feat(button)!: remove click event from button ([495c748](https://github.com/no-gravity-company/no-gravity-elements/commit/495c74885f1da2154281d64c7de0b19cc9758148))

### BREAKING CHANGES

- Remove click event from button

# [1.2.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@1.1.20...@no-gravity-elements/button@1.2.0) (2024-04-01)

### Features

- Add onClick event to nge-button ([8f99196](https://github.com/no-gravity-company/no-gravity-elements/commit/8f991961d28cd97ccf0c95d10fcdef6d47432142))

## [1.1.20](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@1.1.19...@no-gravity-elements/button@1.1.20) (2024-03-30)

**Note:** Version bump only for package @no-gravity-elements/button

## [1.1.19](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@1.1.0...@no-gravity-elements/button@1.1.19) (2024-03-29)

**Note:** Version bump only for package @no-gravity-elements/button

# [1.1.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/button@1.0.8...@no-gravity-elements/button@1.1.0) (2024-03-14)

### Bug Fixes

- Ignore changelogs in lerna config ([9f9c7ee](https://github.com/no-gravity-company/no-gravity-elements/commit/9f9c7ee07e4e05f8dfe9c934bf884515ee8d0732))

### Features

- Add icon to button WC ([b6a75f2](https://github.com/no-gravity-company/no-gravity-elements/commit/b6a75f2d6d07caf3ec001b14ad9cc38d79b2274a))
