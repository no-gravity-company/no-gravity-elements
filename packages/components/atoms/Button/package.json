{
  "name": "@no-gravity-elements/button",
  "version": "2.7.4",
  "description": "No Gravity Element",
  "author": "No Gravity Company",
  "private": false,
  "homepage": "https://github.com/no-gravity-company/no-gravity-elements#readme",
  "license": "ISC",
  "main": "lib/index.js",
  "typings": "types/index.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib/index.js",
    "lib/index.js.map",
    "types/index.ts"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/no-gravity-company/no-gravity-elements.git"
  },
  "scripts": {
    "publish-component": "yarn publish --access=public"
  },
  "bugs": {
    "url": "https://github.com/no-gravity-company/no-gravity-elements/issues"
  }
}
