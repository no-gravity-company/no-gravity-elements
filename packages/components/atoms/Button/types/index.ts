import { ButtonTypes, IconNames } from '@no-gravity-elements/types';

export interface ButtonProps {
  type?: ButtonTypes;
  icon?: IconNames;
}
