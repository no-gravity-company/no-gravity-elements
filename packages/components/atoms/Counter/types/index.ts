export interface CounterProps {
    title: string;
}
