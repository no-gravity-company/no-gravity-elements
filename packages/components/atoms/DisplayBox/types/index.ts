export interface DisplayBoxProps {
    label: string;
    value: number;
}
