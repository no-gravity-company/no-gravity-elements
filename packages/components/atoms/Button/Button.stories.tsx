import { Meta, StoryObj } from '@storybook/web-components';
import { html } from 'lit-html';

import { ButtonProps, ButtonTypes } from '@atoms/Button/types';

import { IconNames } from '@types';

import '@no-gravity-elements/button';

type CustomArgs = ButtonProps & { text: string };

const meta: Meta<CustomArgs> = {
  title: 'atoms/Button',
  component: 'nge-button',
  parameters: {
    webComponents: {
      tagName: 'nge-button',
    },
  },
  argTypes: {
    text: {
      description: 'Custom text',
      control: { type: 'text' },
    },
    type: {
      description: 'Button type',
      options: Object.values(ButtonTypes),
      table: {
        defaultValue: { summary: ButtonTypes.primary },
      },
    },
    icon: {
      description: 'Button Icon name',
      options: [undefined, ...Object.values(IconNames)],
      table: {
        defaultValue: { summary: 'undefined' },
      },
    },
  },
  render: ({ type, text, icon }) =>
    html` <nge-button type=${type} icon=${icon}>${text}</nge-button> `,
};

export default meta;
type Story = StoryObj<CustomArgs>;

export const Default: Story = {
  args: {
    type: ButtonTypes.primary,
    text: 'Button',
    icon: undefined,
  },
};
