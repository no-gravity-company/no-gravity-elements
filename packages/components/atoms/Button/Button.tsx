import { Fragment, FunctionComponent, h } from 'preact';

import { ButtonTypes, IconNames, IconSizes, StringBoolean } from '@no-gravity-elements/types';

import { ButtonProps } from './types';
import classNames from 'classnames';
import './Button.modules.scss';

/**
 * <nge-button>
 *
 *  Button component
 *
 * @element nge-button
 *
 * @prop {ButtonTypes} [type] - Button type
 * @prop {IconNames} [icon] - Button Icon name
 * @prop {boolean} [disabled] - Toggles disabled state
 *
 * @example
 * <nge-button type="primary">Button</nge-button>
 */

const Button: FunctionComponent<ButtonProps> = ({ value, type, icon, disabled = StringBoolean.FALSE }: ButtonProps) => {
  const isDisabled = disabled === StringBoolean.TRUE;
  const buttonClass = classNames((type && ButtonTypes[type]) || ButtonTypes.primary, 'nge-button', {
    disabled: isDisabled,
  });
  console.log(icon)
  return (
    <Fragment>
      <button className={buttonClass} disabled={isDisabled} aria-disabled={isDisabled}>
        {icon && <nge-icon name={icon} size={IconSizes.small} />}
        <span>{value}</span>
      </button>
    </Fragment>
  );
};

Button.observedAttributes = ['type', 'icon', 'disabled'];
Button.useShadowDom = false;

export default Button;
