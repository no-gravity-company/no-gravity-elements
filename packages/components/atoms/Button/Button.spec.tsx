import { h } from 'preact';
import { shallow } from 'enzyme';

import { ButtonTypes } from '@no-gravity-elements/types';

import Button from './Button';

describe('Button', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Button type={ButtonTypes.primary}>Test</Button>);
    expect(wrapper).toMatchSnapshot();
  });

  it.each`
    type
    ${ButtonTypes.secondary}
    ${ButtonTypes.destructive}
    ${undefined}
  `('should render correctly with the provided size and tag', ({ type }) => {
    const wrapper = shallow(<Button type={type}>Button</Button>);
    const element = wrapper.find(`button.${type || 'primary'}`);
    expect(element.exists()).toBe(true);
  });
});
