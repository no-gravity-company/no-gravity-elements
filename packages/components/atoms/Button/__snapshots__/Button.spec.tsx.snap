// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
<button
  aria-disabled={false}
  aria-label="test"
  className="primary nge-button"
  disabled={false}
  type="submit"
>
  <nge-icon
    className="loading-icon"
    name="bouncing-circles"
    size="medium"
  />
  <span
    className="button-text"
  >
    test
  </span>
</button>
`;
